lib/wolfe/cleanup.rb

```diff
@@ -50,31 +51,43 @@ def timespan_regex
       #
 
       def cleanup( config )
-        daily_date = Date.today - eval( config['one_per_day_timespan'] )
-        monthly_date = Date.today - eval( config['one_per_month_timespan'] )
-        first_relevant_date = Date.today - 5.years
+        daily_date = Date.today - eval(config['one_per_day_timespan'])
+        monthly_date = set_monthly_date(config['one_per_month_timespan'])
+        keep_one = !one_per_month_timespan_starts_with_zero?(config['one_per_month_timespan'])
 
         if File.directory?(config['path'])
-          clean_monthly( monthly_date, daily_date, config )
-          clean_yearly( first_relevant_date, monthly_date, config )
+          clean_monthly(monthly_date, daily_date, config, keep_one)
+          clean_yearly(monthly_date, config, keep_one) if keep_one
         else
           puts "Path '#{config['path']}' is not a directory."
         end
       end
 
-      def clean_monthly( monthly_date, daily_date, config )
-        monthly_date.upto( daily_date ) do |date|
-          delete_but_keep_one_per_month( config['path'], config['filename'], date )
+      def set_monthly_date(one_per_month_timespan)
+        if Date.today - eval(one_per_month_timespan) == Date.today
```

Comment:
Ist `eval` wirklich notwendig? Weil: [eval is evil](https://wiki.ruby-portal.de/Evil_eval.html).

Reply:
I understand your concerns. How do you find the new solution?